lib/components/normal-components/Chip.ts

```diff
@@ -52,7 +52,7 @@ export class Chip<PinLabels extends string = never> extends NormalComponent<
       schPinSpacing: pinSpacing,
       schPinStyle: props.schPinStyle,
 
-      pinCount,
+      pinCount: pinCount === 0 ? this.getPortsFromFootprint().length : pinCount,
```

Comment:
can you make the `pinCount` accurate on line 41? Having the condition there makes more sense because we know to trust the `pinCount` variable to represent the `pinCount` at all times